### ISSUE TITLE
linter: `func-names` with `"never"` false positive on TypeScript function overload signatures

### ISSUE THREAD

**Issue by hcwhan** [oxc-project/oxc#24800]
### What version of Oxlint are you using?

1.75.0

### What command did you run?

`oxlint overload.ts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "rules": {
    "func-names": ["error", "never"]
  }
}
```

### What happened?

`func-names` with the `"never"` option reports TypeScript function overload signatures (and `declare function` statements) as "named function expressions", even though they are declarations, not function expressions.

Input (`overload.ts`):

```ts
export function foo(a: string): string
export function foo(a: number): number
export function foo(a: string | number): string | number {
  return a
}
```

Output:

```
overload.ts:1:8: error eslint(func-names): Unexpected named function `foo`. help: Remove the name on this function expression.
overload.ts:2:8: error eslint(func-names): Unexpected named function `foo`. help: Remove the name on this function expression.
```

ESLint with the same config (`func-names: ["error", "never"]` + `@typescript-eslint/parser`) reports no errors on this file, since the ESLint rule only visits `FunctionExpression` nodes, and overload signatures are `TSDeclareFunction` nodes.

The cause seems to be in `is_invalid_function` in `crates/oxc_linter/src/rules/eslint/func_names.rs`, where the `Never` branch approximates "is a function expression" as "is not a function declaration":

```rust
FuncNamesConfigType::Never => {
    func_name.is_some() && func.r#type != FunctionType::FunctionDeclaration
}
```

Overload signatures and `declare function` have `FunctionType::TSDeclareFunction`, so they match this condition and get reported.

Additionally, applying the suggestion (`--fix-suggestions`) removes the name from the overload signatures and produces syntactically invalid code:

```ts
export function (a: string): string
export function (a: number): number
export function foo(a: string | number): string | number {
  return a
}
```

Related precedents where TS overloads caused false positives in other ported rules: #10396 (`no-redeclare`), #5563 (`no-unused-vars`).
